### DISCUSSION TITLE
We need PO editor

### DISCUSSION THREAD

**Discussion by dodyg** [OrchardCMS/OrchardCore#8435] [Ideas]
Right now it is really cumbersome to deal with string translation because you have to deal with source code level to deploy and enrich the localization key/string. 

It will make life so much easier if admin/editor can just add more translation key/string from the admin section and use them immediately at the template.

**Comment by nogginbox** [comment]
Orchard are using [Crowdin](https://crowdin.com/) to manage the localisations. I've been meaning to check it out for another project, but not got round to it yet. You might want to look into this or see if there are other po editor tools on the market.

**Comment by dodyg** [comment]
That's not the problem though. The problem lies in maintaining multi lingual templates. As your multi-lingual page grows, you will need more translation data to be available. Right now you have to deploy to do so or use a bunch of if statement in your liquid template.

**Comment by hishamco** [comment]
> It will make life so much easier if admin/editor can just add more translation key/string from the admin section and use them immediately at the template.

@dodyg the localization load the localizable strings when the `Startup` of the localization module runs, so adding them at runtime needs restart the app. I think you could add your localization strings directly in the PO file, so when the app fires all of your resources will be available

**Reply by dodyg** [reply]
This is not really a practical solution though for bigger installation with multi-tenancy. I mean translation table is a pretty common scenario for multi-lingual websites. There are workarounds for this issue but I think this should be part of a core infrastructure.  Having to restart multi tenant websites in order to enable T["happy-new-year"] in one of the website breaks user experience.

**Reply by hishamco** [reply]
Moving translations to table will change the resources source, I already did somthing similar in the past with `EFLocalizer`, we planned to do something similar but for those localization strings that are dynamic such as content types, content fields .. etc

**Reply by dodyg** [reply]
@hishamco is there any mechanism to reset cached localization string so there is no need to restart the system? I will be working on a PO editor that touch *.po files directly. The question is whether there is a way to let ASP.NET Core knows that these files have been changed and it needs to reload the PO files. 

Can we have LocalizationManager to support something like Reset or Reload method? https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore/OrchardCore.Localization.Core/LocalizationManager.cs

**Reply by hishamco** [reply]
Luckly this working fine with me in JSON Localizer becuase I'm using configuration APIs. What you want could be done using by invalidate the cache which is not supported in the current implementation AFAIK

FYI recently I was working on Data Localization in OCC I faced similar issue, where I need to refesh the cache, simply I remove the cached entries by the key as shown [here](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Modules/blob/7ca66c2d85e762a2470fcaec568f240d4d9675c8/src/OrchardCoreContrib.DataLocalization/Controllers/AdminController.cs#L160)

We may need a method to purge the cache as provider level, @sebastienros your though?

**Reply by dodyg** [reply]
@hishamco what is a JSON Localizer? Is this a new library?

**Reply by hishamco** [reply]
It's a project that I created long time ago, you can find it [here](https://github.com/hishamco/My.Extensions.Localization.Json) I also did an OC implementation in [Orchard Core Contrib repo](https://github.com/orchardcoreContrib/orchardcoreContrib)
